packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -32,11 +32,56 @@ contract PluginSetupProcessor is DaoAuthorizable {
     bytes32 public constant APPLY_UNINSTALLATION_PERMISSION_ID =
         keccak256("APPLY_UNINSTALLATION_PERMISSION");
 
-    struct PluginUpdateParams {
+    /// @notice the struct containing params for the prepare installation function.
+    struct PrepareInstallParams {
+        PluginRepo.Tag versionTag;
+        PluginRepo pluginSetupRepo;
+        bytes data;
+    }
+
+    /// @notice the struct containing params for the apply installation function.
+    struct ApplyInstallParams {
+        PluginRepo.Tag versionTag;
+        PluginRepo pluginSetupRepo;
+        address plugin;
+        PermissionLib.ItemMultiTarget[] permissions;
+    }
+
+    /// @notice the struct containing params for the apply installation function.
+    struct PrepareUpdateParams {
         address plugin; // The address of the `Plugin` contract being currently active. This can be a proxy or a concrete instance.
         PluginRepo pluginSetupRepo; // The repository storing the `PluginSetup` contracts of all versions of a plugin.
-        address currentPluginSetup; // The `PluginSetup` contract of the version being currently active and to be updated from.
-        address newPluginSetup; // The `PluginSetup` contract of the version to be updated to.
+        PluginRepo.Tag currentVersionTag; // The `current` version of the plugin being currently active and to be updated from.
+        PluginRepo.Tag newVersionTag; // The `new` version of the plugin to be updated to.
```

Comment:
Does this allow us to prevent jumping from release1-build-5 to release1-build-2?